pkg/lib/indexer/indexer.go

```diff
@@ -696,3 +698,92 @@ func (i ImageIndexer) DeprecateFromIndex(request DeprecateFromIndexRequest) erro
 
 	return nil
 }
+
+// GraphFromIndexRequest defines the parameters to send to the ExportFromIndex API
+type GraphFromIndexRequest struct {
+	Index         string
+	Packages      []string
+	OutputFile    string
+	ContainerTool containertools.ContainerTool
+	CaFile        string
+	SkipTLS       bool
+}
+
+// GraphFromIndex takes a GraphFromIndexRequest and writes a graphviz DOT file.
+func (i ImageIndexer) GraphFromIndex(request GraphFromIndexRequest) error {
+	databaseFile := request.Index
+	if _, err := os.Stat(request.Index); err != nil {
+		// set a temp directory
+		workingDir, err := ioutil.TempDir("./", tmpDirPrefix)
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(workingDir)
+
+		// extract the index database to the file
+		databaseFile, err = i.getDatabaseFile(workingDir, request.Index, request.CaFile, request.SkipTLS)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Any concern about allowing the user to pass a database file directly?

This helps users avoid waiting for image pulls if they already have a DB file. But it also exposes the DB file as an API in an `index` subcommand.

Reply:
In general opm has two parallel sets of commands: `opm registry` which deals with the db files directly, and `opm index` which deals with images. To keep things consistent, the `opm index graph` should deal with the image, and we can have `opm registry graph` deal with the db.

(low hanging fruit refactoring for this codebase would be to have a single standard index -> db file unwrapper so that you write one registry/index feature and get both commands for free)